src/Void.Core/Filesystem.fs

```diff
@@ -0,0 +1,33 @@
+﻿namespace Void.Core
+
+module Filesystem =
+    open System
+    open System.IO
+
+    type LinesOrFailure =

```

Comment:
In Haskell We'd probably just say `Either [String] Error` at the point of usage.